test/e2e/framework/util.go

```diff
@@ -647,10 +647,7 @@ func WaitForPodsRunningReady(c clientset.Interface, ns string, minPods, allowedN
 			case res && err == nil:
 				nOk++
 			case pod.Status.Phase == v1.PodSucceeded:
-				Logf("The status of Pod %s is Succeeded which is unexpected", pod.ObjectMeta.Name)
-				badPods = append(badPods, pod)
-				// it doesn't make sense to wait for this pod
-				return false, errors.New("unexpected Succeeded pod state")
+				continue
```

Comment:
Isn't this only an issue on startup?  So we can should continue on kube-system, b/c it would check for failed cleanup on overlapping e2e runs.

Reply:
It would be better to add some comment or Logf() to explain why we can skip in this case.